Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -274,67 +300,175 @@ def TestConfigUpdate(Contents):
             retval = -1
     return retval
 
-def SetConfigUpdate(Contents):
-    destFileFullPath = CONFIG_PATH.__add__(DEST_FILE_NAME)
+def TestMultiHoming(ConfigID, Ensure):
+    # Check if state directory exists if Ensure is Present
+    if Ensure == 'Present' and not os.path.exists(PATH_NPM_STATE):
+        LG().Log(LogType.Info, 'Workspace has NPM enabled but state directory absent, set will create')
+        return -1
+
+    # Get workspace lists
+    wsListFromState = GetWorkspaceListFromStateFile()
+    wsListFromDirs = GetWorkspaceListFromDirectories()
+
+    # Check if state file has to be altered
+    for x in wsListFromState:
+        if x not in wsListFromDirs:
+            LG().Log(LogType.Info, 'Workspaces either onboarded/removed, set to rectify state file')
```

Comment:
The intention of the two lists is NOT to have sync between them.
The workspace list from directories would tell the current workspaces onboarded which are for omsagent multihoming.
The workspace list from state file indicate which of those are actually having the NPM solution enabled for the workspace.
NPM solution can be enabled for some and disabled for other workspaces in the multi-homing scenario for omsagent
The check is made to see that we don't have any workspace in state file which is not in the directory list alone.